src/gtk/toga_gtk/font.py

```diff
@@ -32,7 +32,9 @@ def __init__(self, interface):
         try:
             font = _FONT_CACHE[self.interface]
         except KeyError:
-            font = Pango.FontDescription.from_string('{font.family} {font.size}'.format(font=self.interface))
+            font = Pango.FontDescription.from_string(
+                '{font.family} {font.style} {font.variant} {font.weight} {font.size}'.
+                format(font=self.interface))
```

Comment:
I don't know much about GTK or Pango, so I might be completely off here - but is "from_string" the only way to create a Pango definition? Given that you've got a pre-parsed font definition, coercing into a string form so that it can be re-parsed a second time would normally be considered a recipe for unusual side effects.

Reply:
Your concern is valid, and yes there exist setter methods like `set_family`, `set_size`, etc.. but I didn't give them a second thought, since I thought it was reasonable to just update the already existing definition?

Reply:
But you're *not* updating an existing definition - you're creating a new Pango object from a string. I'm asking whether there is a mechanism to create a Pango object *with a specific set of properties*, rather than relying on a parsing-based constructor.

Reply:
If I get you correctly, you're saying it is better to do:
```python
font = pango.FontDescription()
if self.interface.variant == SMALL_CAPS:
   font.set_variant(pango.VARIANT_SMALL_CAPS)
else:
   font.set_variant(pango.VARIANT_NORMAL)
```
and so on, for all the other properties.. right?
Maybe the reason I didn't get you completely at first was due to the fact that all of `toga.fonts` values are compatible with pango's thus the confusion...
Sorry for my ignorance, I hope I got it right this time :cry:

Reply:
That's the general direction I'm suggesting, yes - but what I'm *really* suggesting is the use of a constructor that lets you set those properties *at the time of construction* - `font = pango.FontDescription(variant=pango.VARIANT_SMALL_CAPS, ...)`. Are you saying that Pango has a "from_string" constructor, and an empty constructor, but not a "with arguments" constructor?

Reply:
It is not clear from pygobject's docs but [Gnome Developer's](https://developer.gnome.org/pygtk/stable/class-pangofontdescription.html#constructor-pangofontdescription) docs mention only string as argument and also inspect gives similar result:
```
In [1]: import gi

In [2]: gi.require_version('Pango', '1.0')

In [3]: from gi.repository import Pango

In [4]: import inspect

In [5]: print(inspect.signature(Pango.FontDescription))
(string=None)
```
So, yes that might be the case...